app/src/lib/stores/git-store.ts

```diff
@@ -943,8 +945,13 @@ export class GitStore {
     const pathsToCheckout = new Array<string>()
     const pathsToReset = new Array<string>()
 
+    const submodules = await listSubmodules(this.repository)
+
     await queueWorkHigh(files, async file => {
-      if (file.status !== AppFileStatus.Deleted) {
+      const foundSubmodule = submodules.find(s => s.path === file.path)
```

Comment:
This can be changed to `const submoduleExists = submodules.some(s => s.path === file.path)` and line 352 deleted.